pandas/tests/indexes/common.py

```diff
@@ -682,6 +682,21 @@ def test_hasnans_isnans(self):
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
 
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
+                # Cannot be instantiated from just _data as first argument
+                continue
+
+            data = index._data.copy()
+            idx_type = index.__class__
+            idx = idx_type(data, copy=True)
+            self.assert_numpy_array_equal(data, idx._data)
```

Comment:
hmm, ok then I guess. I don't like reaching in an testing private attributes in tests. How about testing

```
In [4]: arr = ['foo','bar']

In [5]: i = Index(arr)

In [6]: j = Index(i, copy=True)

In [7]: i.values.base
Out[7]: array(['foo', 'bar'], dtype=object)

In [8]: j.values.base
Out[8]: array(['foo', 'bar'], dtype=object)

In [9]: i.values.base is j.values.base
Out[9]: False

In [10]: k = Index(i)

In [11]: k.values.base is i.values.base
Out[11]: True
```

This is a bit more generic and makes sure things are views of the underlying data; we don't actually copy things directly; rather we take views.